sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClientBuilder.java

```diff
@@ -182,23 +182,24 @@ public OpenAIClientBuilder credential(TokenCredential tokenCredential) {
     @Generated private AzureKeyCredential azureKeyCredential;
 
     /** {@inheritDoc}. */
-    @Generated
     @Override
     public OpenAIClientBuilder credential(AzureKeyCredential azureKeyCredential) {
-        this.azureKeyCredential = azureKeyCredential;
-        return this;
+        return this.credential((KeyCredential) azureKeyCredential);
```

Comment:
since we won't be able to drop this method. The workaround approach will be to call the `credential(KeyCredential)` method internally.